### DISCUSSION TITLE
Display the alternate text of a SharePoint Hyperlink field in an Adaptive Card using a slot?

### DISCUSSION THREAD

**Discussion by renwein** [microsoft-search/pnp-modern-search#2837] [Q&A]
Hi All - 

I'm playing with a custom adaptive card layout in the PnP - Search Results web part.  I'm trying to add a hyperlink to a text block, which follows the format of [Title](URL).  I'm using layout slots in my script as follows :

"text": "[${string(jPath($data, concat('.',$root.slots['alertURL']))[0])}}](${string(jPath($data, concat('.',$root.slots['alertURL']))[0])})",

This succeeds in creating the clickable hyperlink in the card, but it is displaying the URL as the clickable text instead of the preferred alternate "Click here to learn more," text. 

Is it possible to tweak the slot string to display the alternate text?  In SharePoint JSON list and library formatting, you can append '.desc' to the end of a field reference to display the alternate text of a hyperlink field; can you do something similar with a slot? Or is there a better way to potentially do this?

Thank you for any suggestions that you can provide.

**Comment by wobba** [comment]
Which field are you using as the URL, and what data does this field contain if you look at the data being returned from search? You can use the debug template to look at the actual data.

**Reply by renwein** [reply]
Thanks, @wobba, for bringing a configuration problem to my attention.  Just looked at the debug template, and the alt text doesn't show up.  We're using a mapped RefinableString property in the slot against a list that was built by someone else, and it's a list-based column instead of a site column.  So we need to go back, swap in a site column, wait for a crawl, and remap the RefinableString to the ows_q_URLH_ColumnName property.

Once that's done, will the alt text appear in the slot in the [Title] position as I've been attempting, or is there still alternate syntax that may be needed to display the URL text?

**Reply by wobba** [reply]
You could check the output of the automatic managed property which already has `ows_q_URLH_ColumnName` mapped and use that one instead as it seems you are using site columns.
